gitlab.go

```diff
@@ -123,7 +123,10 @@ func NewClient(httpClient *http.Client, token string) *Client {
 	}
 
 	c := &Client{client: httpClient, token: token, UserAgent: userAgent}
-	c.SetBaseURL(defaultBaseURL)
+	if err := c.SetBaseURL(defaultBaseURL); err != nil {
+		// should never happen since defaultBaseURL is our constant
+		panic(err)
+	}
```

Comment:
I use panics in my libraries, for example in `mvdan/xurls` where I use `regexp.MustCompile()` which under the covers uses a `panic` if `regexp.Compile()` errors.

Then of course, I have a set of tests that must pass at every commit. These tests actually use the regex, so I know that the library won't panic under any circumstances.

We do want to check for an error. If, for any reason - e.g. the user has a modified version of Go - this fails, we can't just ignore the error and act normally. Whether to panic or to error is more your decision. I think panicking is fine because you are already making sure that this doesn't happen under normal circumstances.

OTOH, if you're happier with returning an error, I can change that.